### PR TITLE
DROOLS-2234 DMN JSON payload with decision-name or decision-id null causes NPE

### PR DESCRIPTION
**Pull request by tarilabs**
/cc @baldimir @etirelli @kurobako

### CODE REVIEW

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/DMNServicesClientImpl.java [hunk 1]
```diff
@@ -111,7 +111,11 @@ public DMNServicesClientImpl(KieServicesConfiguration config, ClassLoader classL
             return evaluateDecisions(containerId, payload);
         }
 
-        private ServiceResponse<DMNResult> evaluateDecisions(String containerId, DMNContextKS payload) {
+        /**
+         * Please notice this method is NOT exposed to the API interface.
+         */
+        // DO NOT ADD @Override
+        public ServiceResponse<DMNResult> evaluateDecisions(String containerId, DMNContextKS payload) {
```
Comment (tarilabs) [inline kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/DMNServicesClientImpl.java]: moved to `pubilc` anyway an instance to the end-user is normally the API interface, which is used by a provisioning mechanism of the general kie-server API.

File: kie-server-parent/kie-server-services/kie-server-services-dmn/src/main/java/org/kie/server/services/dmn/ModelEvaluatorServiceBase.java [hunk 2]
```diff
@@ -137,8 +124,8 @@ public static DMNDecisionInfo decisionToInfo(DecisionNode decisionNode) {
             
             DMNResult result = null;
 
-            final List<String> names = evalCtx.getDecisionNames();
-            final List<String> ids = evalCtx.getDecisionIds();
+            final List<String> names = Optional.ofNullable(evalCtx.getDecisionNames()).orElse(Collections.emptyList());
+            final List<String> ids = Optional.ofNullable(evalCtx.getDecisionIds()).orElse(Collections.emptyList());
```
Comment (tarilabs) [inline kie-server-parent/kie-server-services/kie-server-services-dmn/src/main/java/org/kie/server/services/dmn/ModelEvaluatorServiceBase.java]: the actual fix.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-dmn/src/test/java/org/kie/server/integrationtests/dmn/DMNInputDataStringIntegrationTest.java [hunk 3]
```diff
@@ -185,4 +188,22 @@ public void testInputStringEvaluateDecisionById() {
         }
     }
     
+    @Test
+    public void testDROOLS2234() {
+        // DROOLS-2234
+        DMNContext dmnContext = dmnClient.newContext();
+        dmnContext.set("Full Name", "John Doe");
+
+        DMNServicesClientImpl testClient = (DMNServicesClientImpl) dmnClient;
```
Comment (tarilabs) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-dmn/src/test/java/org/kie/server/integrationtests/dmn/DMNInputDataStringIntegrationTest.java]: downcasted to our internal implementation to use the raw-access method.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-dmn/src/test/java/org/kie/server/integrationtests/dmn/DMNInputDataStringIntegrationTest.java [hunk 4]
```diff
@@ -185,4 +188,22 @@ public void testInputStringEvaluateDecisionById() {
...
+
+        DMNContextKS custom_payload = new DMNContextKS(MODEL_NAMESPACE, MODEL_NAME, dmnContext.getAll());
+        custom_payload.setDecisionIds(null);
+        custom_payload.setDecisionNames(null);
```
Comment (tarilabs) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-dmn/src/test/java/org/kie/server/integrationtests/dmn/DMNInputDataStringIntegrationTest.java]: forging manually the request which when on JSON transport causes the NPE.

**Comment by etirelli** [comment]
Thank you @tarilabs ! Good job!

**Comment by etirelli** [comment]
This needs to be backported to 7.5.x.

**Pull request merged**